### PR TITLE
Simplified semi_circular_loop function in asdf coordinates example

### LINKED ISSUE #4900: Simplify semi_circular_loop function in asdf coordinates example
In the [example that shows how to save a coordinate into an asdf file](https://docs.sunpy.org/en/stable/generated/gallery/saving_and_loading_data/coordinates_in_asdf.html#sphx-glr-generated-gallery-saving-and-loading-data-coordinates-in-asdf-py), there is a function called `semi_circular_loop` that is needlessly complicated.

To make this example more straightforward, it would be better to replace it with,

```python
@u.quantity_input
def semi_circular_loop(length: u.cm, latitude: u.deg = 0*u.deg):
    """
    Return HGS coordinates for a semi-circular loop
    """
    s = np.linspace(0, length, 1000)
    z = length / np.pi * np.sin(np.pi * u.rad * s/length)
    x = np.sqrt((length / np.pi)**2 - z**2)
    x = np.where(s < length/2, -x, x)
    hcc_frame = frames.Heliocentric(
        observer=frames.HeliographicStonyhurst(lon=0 * u.deg, lat=latitude, radius=const.au))
    )
    return SkyCoord(x= x, y=np.zeros(x.shape)*u.cm, z=z + const.R_sun, frame=hcc_frame)
```

### PR DESCRIPTION
**Pull request by 4molybdenum2**
Fixes #4900 

Simplified the semi_circular_loop function in examples/saving_and_loading_data/coordinates_in_asdf.py.
Replaced the current code with the code provided by @ayshih 

```python
@u.quantity_input
def semi_circular_loop(length: u.cm, latitude: u.deg = 0*u.deg):
    """
    Return HGS coordinates for a semi-circular loop
    """
    angles = np.linspace(0, 1, 1000) * np.pi * u.rad
    z = length / np.pi * np.sin(angles)
    x = length / np.pi * np.cos(angles)
    hcc_frame = frames.Heliocentric(
        observer=frames.HeliographicStonyhurst(lon=0 * u.deg, lat=latitude, radius=const.au)
    )
    return SkyCoord(x= x, y=np.zeros_like(x), z=z + const.R_sun, frame=hcc_frame)
```

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @4molybdenum2

**Comment by nabobalis** [comment]
Looks like the docbuild failed with an error:

```python
/home/vsts/work/1/s/examples/saving_and_loading_data/coordinates_in_asdf.py failed leaving traceback:
Traceback (most recent call last):
  File "/home/vsts/work/1/s/examples/saving_and_loading_data/coordinates_in_asdf.py", line 59, in <module>
    loop_coords = semi_circular_loop(500*u.Mm, 30*u.deg)
  File "/home/vsts/work/1/s/.tox/build_docs/lib/python3.8/site-packages/astropy/units/decorators.py", line 251, in wrapper
    return_ = wrapped_function(*func_args, **func_kwargs)
  File "/home/vsts/work/1/s/examples/saving_and_loading_data/coordinates_in_asdf.py", line 48, in semi_circular_loop
    observer=frames.HeliographicStonyhurst(lon=0 * u.deg, lat=latitude, radius=const.au))
NameError: name 'const' is not defined
```

**Comment by 4molybdenum2** [comment]
Sorry but I am not familiar with the code. But do you think this error might have something to do with R_sun?
Like should we replace **R_sun** with **r_sun** like in the original function? Where **r_sun = constants.radius**

**Comment by nabobalis** [comment]
I think you just need to add "from sunpy.sun import constants" and change "const" to "constants" in the code.

**Comment by 4molybdenum2** [comment]
Okay, working on it.

File: examples/saving_and_loading_data/coordinates_in_asdf.py [hunk 1]
```diff
@@ -39,30 +38,21 @@
...
+        observer=frames.HeliographicStonyhurst(lon=0 * u.deg, lat=latitude, radius=constants.au))
 
     return SkyCoord(
-        x=r_2 * np.sin(phi),
-        y=0 * u.cm,
-        z=r_2 * np.cos(phi) + r_sun,
-        frame=hcc_frame).transform_to('heliographic_stonyhurst')
+        x=x,
+        y=np.zeros_like(x),
+        z=z + constants.R_sun,
```
Comment (nabobalis) [inline examples/saving_and_loading_data/coordinates_in_asdf.py:54]:
```suggestion
        z=z + constants.radius,
```

**Comment by nabobalis** [comment]
Thanks again!

**Pull request merged**